untrusted/urts/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+            )
+        };
+        match result {
+            sgx_status_t::SGX_SUCCESS => Ok(Enclave::new(enclave_id)),
+            error => Err(Error::SgxStatus(error)),
+        }
+    }
+}
+
+impl From<Vec<u8>> for EnclaveBuilder {
+    fn from(bytes: Vec<u8>) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes,
+            debug: false,
+        }
+    }
+}
+
+impl Enclave {
+    fn new(id: sgx_enclave_id_t) -> Enclave {
+        Enclave { id }
+    }
+}
+
+impl Deref for Enclave {
+    type Target = sgx_enclave_id_t;
+    fn deref(&self) -> &Self::Target {
+        &self.id
+    }
+}
+
+impl Drop for Enclave {
+    /// Destroys the enclave through the SGX interface.
+    /// Any de-referenced value from [Enclave] is not valid after
+    /// dropping.
+    fn drop(&mut self) {
+        // Per the docs, this will only return SGX_SUCCESS or
+        // SGX_ERROR_INVALID_ENCLAVE_ID. The invalid ID error will only
+        // happen when the ID is invalid, the enclave hasn't been loaded,
+        // or the enclave has already been destroyed. Any of these cases
+        // don't afford corrective action, so ignore the return value
+        unsafe { sgx_destroy_enclave(self.id) };
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use test_enclave::{ecall_add_2, ENCLAVE};
+
+    #[test]
+    fn fail_to_create_enclave_with_non_existent_file() {
+        let builder = EnclaveBuilder::new(b"garbage bytes");
+        assert_eq!(
+            builder.create(),
+            Err(Error::SgxStatus(sgx_status_t::SGX_ERROR_INVALID_ENCLAVE))
+        );
+    }
+
+    #[test]
+    fn creating_enclave_succeeds() {
+        let builder = EnclaveBuilder::new(ENCLAVE);
+        assert!(builder.create().is_ok());
+    }
+
+    #[test]
+    fn create_enclave_builder_from_vector() {
+        let vector = ENCLAVE.to_vec();
+        assert!(EnclaveBuilder::from(vector).create().is_ok());
+    }
+
+    #[test]
+    fn calling_into_an_enclave_function_provides_valid_results() {
+        // Note: the `debug()` was added to ensure proper builder behavior of
+        // the `create()` method.  It could go away if another test has need
+        // of similar behavior.
+        let enclave = EnclaveBuilder::new(ENCLAVE).debug(true).create().unwrap();
+
+        let mut sum: c_int = 3;
+        let result = unsafe { ecall_add_2(*enclave, 3, &mut sum) };
+        assert_eq!(result, sgx_status_t::SGX_SUCCESS);
+        assert_eq!(sum, 3 + 2);
+    }
+
+    #[test]
+    fn default_debug_flag_is_0() {
+        // For the debug flag it's not easy, in a unit test, to test it was
+        // passed to `sgx_create_enclave()`, instead we focus on the
+        // `as c_int` portion maps correctly to 0 or 1
+        let builder = EnclaveBuilder::new(b"");
+        assert_eq!(builder.debug as c_int, 0);
+    }
+
+    #[test]
+    fn when_debug_flag_is_true_it_is_1() {
+        let builder = EnclaveBuilder::new(b"");
+        let builder = builder.debug(true);
```

Comment:
Optional/Equivalent: Merge these declarations
```suggestion
        let builder = EnclaveBuilder::new(b"").debug(true);
```